### ISSUE TITLE
Investigate learning-based approach for human base estimation without contact assumption

### ISSUE THREAD

**Issue by Zweisteine96** [ami-iit/element_physics-neural-inertial-odometry#9]
As stated in this issue: https://github.com/ami-iit/element_physics-neural-inertial-odometry/issues/8, I'll document the investigation of a learning-based approach for human base estimation here.

**Comment by Zweisteine96** [comment]
### What to learn
Let's say we are interested at the following quantities of the base expressed in the World frame: 
- global position $p_b$
- global orientation $R_b$
- global linear velocity $v_b$

We have the inertial measurements at different levels as potential inputs:
- Minimum level: one single IMU attached to the Pelvis, which provides the acceleration $a_s$ and angular velocity $w_s$ expressed in the sensor frame.
- Full level: one IMU attached to Pelvis, two IMUs attached to left/right forearms, two IMUs attached to left/right lower legs, which provide as well $a_s$ and $w_s$.

Additionally, we can also have the full-body joint kinematics as potential inputs. (Depending on the human modeling, for example, `URDF` or `SMPL`, the joint kinematics can be represented in different ways)

> NOTE: we have neither the contact assumption (at any timestamp, there is always at least one foot-ground contact available) nor the zero-velocity assumption (any non-slipping contact foot with the ground has zero velocity).

**Comment by Zweisteine96** [comment]
### How to learn
How can we learn the desired kinematics of the floating base in the global world from the limited available measurements and observations?

#### Supervised learning
A very straightforward solution could be `supervised learning`. We can choose and process the input and output features as we want. We can collect some (maybe more than just some) synchronized data of IMUs (probably from Xsens or iFeel) and ground truth of base (probably from a Vicon system). We can try different network architectures and do all the lego stuffs, namely pick one structure from here and another structure from there. We can define as many loss functions as we want. We can finally train the network using all the tricks we can try. 

And then? We might still face the generalization problem as all the SL-based methods can suffer. And obviously there is something lack here, the NOVELTY. Everything just seems too common so far, and we can not even guarantee the final performance.

**Comment by Zweisteine96** [comment]
### Can we explore reinforcement learning as a state estimator?

Our ultimate goal is to know the base state at any current timestamp $t$, given the base state at the last timestamp $t-1$ (or even a sequence of past states from $t-N$ to $t-1$), and a set of inertial measurements at current timestamp $t$. More specifically, the relationship between the base state and the inertial measurements can be described with IMU kinematics equations. This is widely used as the motion model (or propagation/advancement step) in a filtering-based method. 

Can we regard this as a Markov decision process? The measurements are considered as the `actions` $a_t$ to derive the next state $s_t$ from the last state $s_{t-1}$.

**Comment by Zweisteine96** [comment]
Save a nice work here: [Deep reinforcement learning for modeling human locomotion control in neuromechanical simulation](https://link.springer.com/content/pdf/10.1186/s12984-021-00919-y.pdf)

**Comment by Zweisteine96** [comment]
Some slides of my thoughts on using RL as a state estimator: https://istitutoitalianotecnologia-my.sharepoint.com/:p:/r/personal/cheng_guo_iit_it/Documents/[RL4BaseEstimation.pptx](https://istitutoitalianotecnologia-my.sharepoint.com/:p:/r/personal/cheng_guo_iit_it/Documents/RL4BaseEstimation.pptx?d=wa87da52aef6444359a9e3541b2d94a10&csf=1&web=1&e=m6cIbT)?d=wa87da52aef6444359a9e3541b2d94a10&csf=1&web=1&e=m6cIbT

**Comment by qingxinxia** [comment]
Hi @Zweisteine96 

Thank you for your fancy idea, I want to explore it more. The slides link seems broken, could you please check it? Thanks a lot.

**Comment by Zweisteine96** [comment]
> Hi [@Zweisteine96](https://github.com/Zweisteine96)
> 
> Thank you for your fancy idea, I want to explore it more. The slides link seems broken, could you please check it? Thanks a lot.

Hey,

thanks for your interest in this project! I'm sorry that the slides were for internal discussions which can not be made public. But if you are working (or plan to) on something similar, I'm open to any discussions or potential collaborations, you can reach me via gcleon96@gmail.com :)

Best
